pkg/cloudprovider/aws/factory.go

```diff
@@ -86,9 +87,24 @@ func (f *Factory) CapacityFor(provisioner *v1alpha1.Provisioner) cloudprovider.C
 	}
 }
 
+// Checks for env vars in the defined order and if none of those are set falls back to using EC2
+// IMDS for region lookup if neither AWS_DEFAULT_REGION or AWS_REGION is defined.
 func withRegion(sess *session.Session) *session.Session {
+	if region, envAwsDefaultRegion := os.LookupEnv("AWS_DEFAULT_REGION"); envAwsDefaultRegion {
+		log.PrettyInfof("Found AWS_DEFAULT_REGION env var set, using: %v", region)
+		sess.Config.Region = aws.String(region)
+		return sess
+	}
+
+	if region, envAwsRegion := os.LookupEnv("AWS_REGION"); envAwsRegion {
```

Comment:
I'm wary of supporting this legacy field, as it can confuse users and is supported inconsistently. Given that we made this explicit in the helm chart, I'd rather just support one flag.